### PR TITLE
Remove client-go/util/testing/tmpdir.go

### PR DESCRIPTION
**Pull request by shuheiktgw**
#### What type of PR is this?

/kind cleanup

#### What this PR does / why we need it:
This is a part of util package for test clean up.

#### Which issue(s) this PR fixes:

Relates to https://github.com/kubernetes/kubernetes/issues/117385

#### Does this PR introduce a user-facing change?

```release-note
None
```

#### Additional documentation e.g., KEPs (Kubernetes Enhancement Proposals), usage docs, etc.:

```docs
None
```

### CODE REVIEW

File: pkg/controller/volume/attachdetach/util/util_test.go [hunk 1]
```diff
@@ -283,7 +282,7 @@ func Test_CreateVolumeSpec(t *testing.T) {
 }
 
 func setup(nodeName string, t *testing.T) (*volume.VolumePluginMgr, csimigration.PluginManager, csitrans.CSITranslator, fakeframework.PersistentVolumeLister, fakeframework.PersistentVolumeClaimLister) {
-	tmpDir, err := utiltesting.MkTmpdir("csi-test")
+	tmpDir, err := os.MkdirTemp("", "csi-test")
```
Comment (cbandy) [inline pkg/controller/volume/attachdetach/util/util_test.go:285]: Should these `MkdirTemp(); if err, t.Fatal; defer os.RemoveAll` patterns be replaced with [`testing.T.TempDir`](https://pkg.go.dev/testing#T.TempDir)?

The five lines here can be replaced with one: `tmpDir := t.TempDir()`.

Comment (shuheiktgw) [reply pkg/controller/volume/attachdetach/util/util_test.go:285]: Good catch, thank you! Let me fix that 🙂

**Comment by shuheiktgw** [comment]
I'll check the e2e test failure later

**Comment by shuheiktgw** [comment]
BTW, there are so many places that can be cleaned up with `t.TempDir` so I'll create another issue once this PR has been merged 🙂

**Comment by bart0sh** [comment]
@shuheiktgw If you still need this PR then please rebase, if not, please close the PR

**Comment by shuheiktgw** [comment]
I don't have any strong reasons to need to merge this PR so I'll simply close this PR

**Pull request closed without merging**